backend/server/routers/programs.py

```diff
@@ -195,28 +195,27 @@ def add_specialisation(structure: dict, code: str):
 def get_structure(
     programCode: str, spec: Optional[str] = None
 ):
-    structure = {}
-    if spec:
-        specs = spec.split("+") if "+" in spec else [spec]
-        for m in specs:
-            add_specialisation(structure, m)
-
-    # add details for program code
-    programsResult = programsCOL.find_one({"code": programCode})
-    if not programsResult:
-        raise HTTPException(
-            status_code=400, detail="Program code was not found")
-
-    structure['General'] = {}
-    structure['Rules'] = {}
-    with suppress(KeyError):
-        for container in programsResult['components']['non_spec_data']:
-            add_subgroup_container(structure, "General", container, [])
-    apply_manual_fixes(structure, programCode)
-
+    structure = add_specialisations({}, spec)
+    uoc = add_program_code_details(structure, programCode)
+    course_list_from_structure(structure)
```

Comment:
we must still apply manual fixes here!

Reply:
The add program codes function applies manual fixes